### PR TITLE
Add a limit order test

### PR DESCRIPTION
**Pull request by eric-zhu-uw**
## Describe your changes and provide context
- add a dex module test for multiple matching limit orders

### CODE REVIEW

File: x/dex/module_test.go [hunk 1]
```diff
@@ -170,6 +170,163 @@ func TestEndBlockMarketOrder(t *testing.T) {
...
+	_, found = dexkeeper.GetLongBookByPrice(ctx, contractAddr.String(), sdk.MustNewDecFromStr("2"), pair.PriceDenom, pair.AssetDenom)
+	require.False(t, found)
+	_, found = dexkeeper.GetLongBookByPrice(ctx, contractAddr.String(), sdk.MustNewDecFromStr("1"), pair.PriceDenom, pair.AssetDenom)
+	require.True(t, found)
+	_, found = dexkeeper.GetShortBookByPrice(ctx, contractAddr.String(), sdk.MustNewDecFromStr("3"), pair.PriceDenom, pair.AssetDenom)
+	require.True(t, found)
+
+	matchResults, _ := dexkeeper.GetMatchResultState(ctx, contractAddr.String(), 2)
+	require.Equal(t, 2, len(matchResults.Orders))
+	require.Equal(t, 4, len(matchResults.Settlements))
```
Comment (LCyson) [inline x/dex/module_test.go:306]: can we check if correct amount of price=3 entries are remaining here

File: x/dex/module_test.go [hunk 2]
```diff
@@ -170,6 +170,163 @@ func TestEndBlockMarketOrder(t *testing.T) {
...
+			PositionDirection: types.PositionDirection_LONG,
+			Data:              "{\"position_effect\":\"Open\",\"leverage\":\"1\"}",
+		},
+	)
+
+	ctx = ctx.WithBlockHeight(3)
+	testApp.EndBlocker(ctx, abci.RequestEndBlock{})
+
+	matchResults, _ = dexkeeper.GetMatchResultState(ctx, contractAddr.String(), 3)
+	require.Equal(t, 1, len(matchResults.Orders))
```
Comment (LCyson) [inline x/dex/module_test.go:335]: ditto

**Comment by LCyson** [comment]
lgtm, pending nit comments

File: x/dex/module_test.go (hunk 1 above)
Comment (eric-zhu-uw) [reply x/dex/module_test.go:306]: sounds good, added one for the long book at price=3, there already is a check for the short book at price=3.

**Review by LCyson — APPROVED** [review]

**Pull request merged**